src/reflect/scala/reflect/internal/SymbolTable.scala

```diff
@@ -52,6 +52,8 @@ abstract class SymbolTable extends macros.Universe
                               with Reporting
 {
 
+  object SymbolTableLock
```

Comment:
10 lines bellow we have `synchronizeSymbolsAccess`
I think we need to investigate/discuss if it shouldn't be the same lock.
At least we need to be sure we understand logic here.
But if you already do, could you please add comment here elaborating on design decisions?

Reply:
I just missed it - and I am switching to synchronizeSymbolsAccess